### ISSUE TITLE
Error: Unexpected value 'CKEditorModule' imported by the module 'AppModule'

### ISSUE THREAD

**Issue by InigoWD** [chymz/ng2-ckeditor#47]
I have a new project created with the angular cli. I have installed this package using `npm install ng2-ckeditor` and set up my system.config.js and app.module.ts files as per the example. I have run ng build and ng serve. The module is not loading, as I am getting the above error in my console?

**Comment by chymz** [comment]
Hi !
What is your angular-cli version ? The latest uses webpack...

**Comment by InigoWD** [comment]
Ah, OK. Yes, I guess I'm using the webpack version- `ng -v` gives `angular-cli: 0.1.0`

Sorry I'm new to Angular 2. Is there any easy workaround...?

**Comment by chymz** [comment]
`0.1.0` ? the latest version of angular-cli is `1.0.0-beta.17` !

Simple workaround : update your angular-cli :smile: or do without and use SystemJS

**Comment by Rogue85** [comment]
My angular-cli version is `1.0.0-beta.17` and i have a same error. Please, help.

**Comment by InigoWD** [comment]
I don't understand why it says 0.1.0 (perhaps this refers to something else) but I'm pretty sure I'm running the latest version, as I have uninstalled and reinstalled, ie:

`npm uninstall -g angular-cli`
`npm cache clean`
`npm install -g angular-cli@webpack`

I still get the same error

**Comment by chymz** [comment]
Yep I see this error too... webpack seems to bundle ng2-ckeditor correctly.
I'm not very familiar with webpack but i'll take a look

**Comment by uNki23** [comment]
same error here, using angular-cli 1.0.0-beta.17 
should work with angular-cli, since thats the way to go :)

**Comment by devCrossNet** [comment]
@InigoWD install it like this: `npm install -g angular-cli` or `npm install -g angular-cli@latest`. I use CKEditor with angular-cli latest and it works, it also works with --aot.

**Comment by InigoWD** [comment]
@devCrossNet Just to be clear, you use the latest angular cli with *this package*, `ng2-ckeditor`, and it works? I have installed exactly as you say, using `npm install -g angular-cli`. It is using webpack. I get the error given.

**Comment by devCrossNet** [comment]
@InigoWD yes, that is correct. But I use the latest version of the CLI from the master branch locally. I have this version of ng2-ckeditor: `"ng2-ckeditor": "^1.1.3",` via `npm install ng2-ckeditor --save` and use it this way:

> import { CKEditorModule } from 'ng2-ckeditor';
> ...
> @NgModule({
>  imports: [
>    FormsModule,
>    CommonModule,
>    CKEditorModule
>  ],
>  declarations: [...],
>  exports: [...]
>})

**Comment by chymz** [comment]
@InigoWD how did you to make it work ? I've same error as above

It seems a problem with `ngc`. I've tested some fixes but none resolve this problem.
If anyone have a solution, i'll take it :)

**Comment by devCrossNet** [comment]
@chymz: I just installed it and used it. I also have a custom version of ckeditor in my assets directory, but this should not be related to this issue.

But as I said, I use a local version of the master branch of the cli.

<img width="430" alt="bildschirmfoto 2016-10-20 um 16 38 51" src="https://cloud.githubusercontent.com/assets/1667598/19564529/2371ac6a-96e4-11e6-8d2e-23af7e22bc85.png">

AOT:
<img width="618" alt="bildschirmfoto 2016-10-20 um 16 47 25" src="https://cloud.githubusercontent.com/assets/1667598/19564787/f9504ee0-96e4-11e6-8bb0-45676becdfaa.png">

**Comment by chymz** [comment]
Thanks @devCrossNet !
I confirm this module works with angular-cli from master branch.

**Comment by InigoWD** [comment]
Well I still get the same error.

I am a noob to angular2 so I can only assume I'm missing some vital step. Sorry to pester you with what are no doubt stupid questions, but I have found getting started with ng2 to be an absolute nightmare as tutorials and blog posts only a few months old are already woefully out-of-date. It took me ages to understand the changes incurred by the move to webpack. In fact, I still don't fully understand. For example, the step relating to SystemJS Config given on [the home page of this plugin](https://github.com/chymz/ng2-ckeditor); do I need to replicate this somehow/somewhere with the webpack version?

**Comment by devCrossNet** [comment]
@InigoWD you don't need the system-js config if you use angular cli > 1.0.0 beta.14. webpack manages everything for you. but obviously there is a bug in the latest cli version. you can wait for a new release of the cli and everything will work out of the box.

Or you can use a local copy of the cli master branch, there is this bug already fixed. Here is some information how you can install it: https://github.com/angular/angular-cli#development-hints-for-hacking-on-angular-cli

**Comment by InigoWD** [comment]
Thanks for the clarification re. system-js config / webpack, and for the link ^. I've followed the instructions there and can confirm I'm running angular cli 1.0.0-beta.18. I have made a completely new test project using `ng new` and installed the package with `npm ng2-ckeditor`.

Imported into main module, and template used in component as per home page example.

Same problem.

**Comment by InigoWD** [comment]
I beg your pardon, this is working now :)

**Comment by chymz** [comment]
@InigoWD : great ! Can you share your solution to others dudes if you can ;)

**Comment by InigoWD** [comment]
The solution is to use the very latest cli as @devCrossNet describes. Not entirely clear why it wasn't working for me yesterday tbh, I can only think `ng build` / `ng serve` wasn't quite doing the job properly for some strange reason. Nothing really I can add other than to just get a clean project and the latest cli

**Comment by uNki23** [comment]
hmm... i don't get it to work properly.
i npm installed `angular-cli@1.0.0-beta.18` and `ng2-ckeditor@1.1.4`.
i'm on `angular@2.1.1`
running `ng serve` finishes with no errors and valid packages.
in my `app.module.ts` i `import { CKEditorModule } from 'ng2-ckeditor'`.
running my app in the browser it shows `metadata_resolver.js:227Uncaught Error: Unexpected value 'CKEditorModule' imported by the module 'AppModule'`

any ideas? :/

**Comment by ervinn** [comment]
I had the same issue. It is working now.
It is not enough to use 'npm install -g angular-cli@latest' or do the 'npm link'.
Make sure that in your project 'package.json' file you do NOT have a early version specified. 
I did all suggested above, but it was still not working.
After I changed my 'package.json' file using "angular-cli": "1.0.0-beta.18"; after that change it was working, I did not get the error.

`"devDependencies": {
    "concurrently": "^2.0.0",
    "@types/jasmine": "^2.2.30",
    "angular-cli": "1.0.0-beta.**18**",
    "codelyzer": "~0.0.26",
    "jasmine-core": "2.4.1",
    "jasmine-spec-reporter": "2.5.0",`
